clang/lib/Driver/ToolChains/XOCC.cpp

```diff
@@ -215,11 +215,35 @@ void SYCL::LinkerXOCC::constructSYCLXOCCCommand(
       options::OPT_Xsycl_linker_EQ, TC.getTriple(), C.getDriver());
 
   switch (TC.getTriple().getSubArch()) {
-    case llvm::Triple::FPGASubArch_hw: CmdArgs.push_back("hw"); break;
-    case llvm::Triple::FPGASubArch_hw_emu: CmdArgs.push_back("hw_emu"); break;
-    case llvm::Triple::FPGASubArch_sw_emu: CmdArgs.push_back("sw_emu"); break;
-    default:
-      llvm_unreachable("invalid subarch");
+  case llvm::Triple::FPGASubArch_hw:
```

Comment:
Probably in the future we should rename them to `FPGASubArchHlsHw`? But this shows the limitations of CamelCase... :-(